### DISCUSSION TITLE
.NET 6 support

### DISCUSSION THREAD

**Discussion by dzhenko-m** [vcsjones/AzureSignTool#160] [General]
Hi. 
In the latest release it says "Moved off of .NET Core 2.1 to .NET Core 3.1. AzureSignTool will moved to .NET 6.0 LTS later this year."
But it's 2022 already, and there is still no new releases. When is it planed to be released?

**Comment by fatihkaplan** [comment]
Can we get an answer on this please? .Net Core 3.1 is going out of support on Dec 13, 2022. https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-and-net-core

**Comment by vcsjones** [comment]
I hope to cut a release in the next few weeks (`main` is already targeting .NET 6). The latest I expect it will be sometime in late October.

**Comment by fatihkaplan** [comment]
Any luck with cutting a release targeting .NET 6 in these days? If not, and if I were to use /main, is it fully backwards compatible? Thank you!

**Comment by vcsjones** [comment]
Thanks, yes sorry it took a little longer due to some personal reasons.

One that I do want to highlight is that the tool _today_ works with .NET 6. The currently released version has a major roll forward:

https://github.com/vcsjones/AzureSignTool/blob/v3.0.0/src/AzureSignTool/runtimeconfig.template.json

You can completely remove .NET Core 3.1. AzureSignTool uses whatever the latest runtime available you have is.

<img width="1164" alt="image" src="https://user-images.githubusercontent.com/361677/206238760-55d392f5-682f-4b5a-b096-72b4a5d495a8.png">

Note that I am not using this as an excuse to not put out an update, and I intend to do so very soon. But if you are itching to stop installing .NET Core 3.1 today, you can.

**Reply by fatihkaplan** [reply]
Np, and thank you, that's great to hear! I'm not itching really, but company policies concerning security practices forces us to remove unsupported versions from Dev/DevOps environments. Once again, thank you, Kevin! At TechSmith, we've been using AzureSignTool in our CI/CD pipelines successfully with EV certs in Azure premium key vault and have code around it.

**Comment by vcsjones** [comment]
AzureSignTool 4.0.1 is out which is built against .NET 6. Thanks y'all for the patience.
